core/src/com/mygdx/game/steerers/FollowPathSteerer.java

```diff
@@ -159,43 +269,36 @@ public void draw(GameRenderer gameRenderer) {
 		if (pathToRender.size > 0 && currentSegmentIndex >= 0) {
 			MyShapeRenderer shapeRenderer = gameRenderer.shapeRenderer;
 			shapeRenderer.setProjectionMatrix(gameRenderer.viewport.getCamera().combined);
-			shapeRenderer.begin(MyShapeRenderer.ShapeType.Line);
+
+			// Draw path target position
+			Vector3 t = gameRenderer.vTmpDraw1.set(followPathSB.getInternalTargetPosition());
+			t.y -= steerableBody.halfExtents.y;
+			float size = .05f;
+			float offset = size / 2;
+			shapeRenderer.begin(MyShapeRenderer.ShapeType.Filled);
 			shapeRenderer.setColor(Color.CORAL);
+			shapeRenderer.box(t.x - offset, t.y - offset, t.z + offset, size, size, size);
 
-			// Path
-			Vector3 q;
-			Vector3 p = getLinePathPosition(gameRenderer.vTmpDraw1, gameRenderer.vTmpDraw2);
-			for (int i = getCurrentSegmentIndex() + 1; i < pathToRender.size; i++) {
-				q = pathToRender.get(i);
+			// Draw path
+			shapeRenderer.set(MyShapeRenderer.ShapeType.Line);
+			Vector3 p = t;
+			int i = getCurrentSegmentIndex() + 1;
+			if (i + 1 < pathToRender.size && linePath.calculatePointSegmentSquareDistance(gameRenderer.vTmpDraw2, pathToRender.get(i), pathToRender.get(i+1), p) < 0.0001)
```

Comment:
Not sure what it actually does. If it is needed it should be rewritten at some point to be understandable.